### ISSUE TITLE
--config command does not work

### ISSUE THREAD

**Issue by mkilpatrick** [TobyAndToby/generate-license-file#217]
How can I use the beta config options?

```
$ npx generate-license-file --config .glf.json
✖ unknown or unexpected option: --config
```

It's not listed on the [docs](https://generate-license-file.js.org/docs/cli) page, but is on the Github README. According to the docs it should just pick up the file automatically if it's not a custom name, but that doesn't appear to work either. Additionally, the docs say that --input or --output are required, otherwise there will be a prompt, but if there's a config file then the prompt shouldn't happen.

**Comment by mkilpatrick** [comment]
Oh, it's part of [v3.0.0-beta.1](https://github.com/TobyAndToby/generate-license-file/releases/tag/v3.0.0-beta.1)

**Comment by mkilpatrick** [comment]
I am getting errors using it though, both in JSON and YAML.

.glf.yaml
```
inputs:
  - ./package.json
output: ./THIRD-PARTY-NOTICES
append:
  - ./additional-third-party-context.txt
overwrite: true
```
```
⠋ Resolving licenses...{
  inputs: [ './package.json' ],
  output: './THIRD-PARTY-NOTICES',
  overwrite: true,
  append: [
    '/Users/mkilpatrick/Desktop/cli/additional-third-party-context.txt'
  ]
}
⠦ Resolving licenses.../Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/generate-license-file@3.0.0-beta.1_commander@11.1.0/node_modules/generate-license-file/src/lib/utils/packageJson.utils.js:9
        throw new Error(`Cannot find the file: '${pathToPackageJson}'`);
```

.glf.json
```
{
  "inputs": [
    "./package.json"
  ],
  "output": "./THIRD-PARTY-NOTICES",
  "append": [
    "./additional-third-party-context.txt"
  ],
  "overwrite": true
}
```
```
> generate-license-file

/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/json5@2.2.3/node_modules/json5/lib/parse.js:1110
    const err = new SyntaxError(message)
                ^

SyntaxError: JSON5: invalid character 'U' at 1:2
```

**Comment by tobysmith568** [comment]
Hey @mkilpatrick - I'm sorry that we've not replied to this issue for over three weeks 😅
It apparently slipped through the cracks....

To be honest we've never tinkered with pnpm before so our V3 re-write (that's currently in beta) wasn't written with pnpm in mind.
That being said we've got a solution that we believe works great in a draft PR ( #225 ).

@tobybessant and I will review it together over the next few days and get it into a place where it's good to be merged but I wanted to reply to you here to say that we've seen this and are working on it :)

**Comment by mkilpatrick** [comment]
Thanks for the response! I implemented a workaround to accomplish the same thing for the time being, but having it native to this library will be great. Is there a timeframe of moving these new features out of beta?

**Comment by tobysmith568** [comment]
Once @tobybessant and I are happy with PR #225 and it gets merged then we'll probably release a 3.0.0-beta.2 where we'd appreciate it if you could test it locally to see if it fixes your issue. Assuming your issue is fixed then I'd say that a 3.0.0 release can follow right away.

**Comment by tobysmith568** [comment]
Hey @mkilpatrick ,

We've now released version 3.0.0-beta.2 which we believe will fix your issues while using
`json` or `yaml` config files. Would you have some availability to install this version locally and see if it works for you?

If it does then we're ready to release 3.0.0 🚀

**Comment by mkilpatrick** [comment]
I'm getting the same error as my original post using json when using 3.0.0-beta.2.

package.json script
`"generate-notices": "generate-license-file",`

.glf.json
```
{
  "inputs": [
    "./package.json"
  ],
  "output": "./THIRD-PARTY-NOTICES",
  "append": [
    "./additional-third-party-content.txt"
  ],
  "overwrite": true
}
```

Error
```
$ pnpm generate-notices

> @yext/cli@1.0.0 generate-notices /Users/mkilpatrick/Desktop/cli
> generate-license-file

/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/json5@2.2.3/node_modules/json5/lib/parse.js:1110
    const err = new SyntaxError(message)
                ^

SyntaxError: JSON5: invalid character 'U' at 1:2
    at syntaxError (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/json5@2.2.3/node_modules/json5/lib/parse.js:1110:17)
    at invalidChar (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/json5@2.2.3/node_modules/json5/lib/parse.js:1055:12)
    at Object.comment (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/json5@2.2.3/node_modules/json5/lib/parse.js:184:15)
    at lex (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/json5@2.2.3/node_modules/json5/lib/parse.js:100:42)
    at Object.parse (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/json5@2.2.3/node_modules/json5/lib/parse.js:25:17)
    at json5Parse (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/generate-license-file@3.0.0-beta.2_commander@11.1.0_typescript@5.2.2/node_modules/generate-license-file/src/lib/cli/config/parsers/json5.js:6:39)
    at #loadConfiguration (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/cosmiconfig@8.3.6_typescript@5.2.2/node_modules/cosmiconfig/dist/Explorer.js:92:30)
    at #readConfiguration (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/cosmiconfig@8.3.6_typescript@5.2.2/node_modules/cosmiconfig/dist/Explorer.js:77:80)
    at async search (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/cosmiconfig@8.3.6_typescript@5.2.2/node_modules/cosmiconfig/dist/Explorer.js:44:40)
    at async Explorer.search (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/cosmiconfig@8.3.6_typescript@5.2.2/node_modules/cosmiconfig/dist/Explorer.js:71:20) {
  lineNumber: 1,
  columnNumber: 2,
  filepath: '/Users/mkilpatrick/Desktop/cli/.glf.json'
}

Node.js v18.18.2
```

**Comment by tobysmith568** [comment]
Ah that's annoying, I'm going to re-open this issue.

While we look into this, would it be possible for you to share your `package.json` and pnpm lock file with us so that we can get an accurate reproduction? However, if it's closed source or if you use closed source dependencies would you be able to see if you could create a small repro `package.json` and lock file for us? This could potentially speed up our investigation.

Either way, we'll get back to you when we have something :)

**Comment by tobysmith568** [comment]
In fact, don't worry about getting a repro for now because I suspect I've fixed this with PR #235 . It's not merged yet but it will be asap once @tobybessant and I are happy with it.

**Comment by mkilpatrick** [comment]
My package.json has `"license": "BSD-3-Clause",`

**Comment by tobybessant** [comment]
Hey @mkilpatrick,

We've released `v3.0.0-beta.3`. We believe it addresses your issues with JSON and YAML config files - if you get a chance it would be great if you could check it out locally and let us know?

Thanks for your patience 😅

**Comment by mkilpatrick** [comment]
Still no go with beta.3

```
> @yext/cli@1.0.0 generate-notices /Users/mkilpatrick/Desktop/cli
> generate-license-file

node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module 'tslib'
Require stack:
- /Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/generate-license-file@3.0.0-beta.3/node_modules/generate-license-file/src/lib/cli/index.js
- /Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/generate-license-file@3.0.0-beta.3/node_modules/generate-license-file/bin/generate-license-file
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/generate-license-file@3.0.0-beta.3/node_modules/generate-license-file/src/lib/cli/index.js:4:17)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/generate-license-file@3.0.0-beta.3/node_modules/generate-license-file/src/lib/cli/index.js',
    '/Users/mkilpatrick/Desktop/cli/node_modules/.pnpm/generate-license-file@3.0.0-beta.3/node_modules/generate-license-file/bin/generate-license-file'
  ]
}

Node.js v18.18.2
```

**Comment by tobysmith568** [comment]
Hmm, okay, well at least the previous issues are gone and we're making _some_ progress!

I believe I've now resolved your latest issue. In short, I believe it to be caused by a breaking change in our build tool that we were not aware of. `3.0.0-beta.4` has now been released, if you get a sec, we'd appreciate you giving it a go :)

**Comment by mkilpatrick** [comment]
It works 🙌

**Comment by tobysmith568** [comment]
Hey, that's great news @mkilpatrick!

As of today we've released 3.0.0. If you do happen to notice anything then please feel free to raise a new issue :)
